discovery-provider/nginx_conf/main.lua

```diff
@@ -13,6 +13,43 @@ math.randomseed(string.byte(resty_random.bytes(1)))
 
 local _M = {}
 
+local redirect_target_weights = {}
+for index, redirect_target in ipairs(config.redirect_targets) do
+    redirect_target_weights[redirect_target] = 1
+end
+
+function update_redirect_target_weights (premature)
+    if premature then
+        return
+    end
+
+    local maximum_request_count = 0
+    local httpc = resty_http.new()
+    for index, redirect_target in ipairs(config.redirect_targets) do
+        local res, err = httpc:request_uri(redirect_target .. "/health_check", { method = "GET" })
+
+        if not res then
+            ngx.log(ngx.ERR, "failed to make request to discovery provider: ", redirect_target, err)
+            redirect_target_weights[redirect_target] = 0
+        else
+            local request_count = tonumber(res.body)
+            redirect_target_weights[redirect_target] = request_count
+            maximum_reqeust_count = math.max(maximum_request_count, request_count)
+        end
+    end
+    httpc:close()
+
+    for index, redirect_target in ipairs(config.redirect_targets) do
+        if redirect_target_weights[redirect_target] ~= 0 then
+            redirect_target_weights[redirect_target] = (2 * maximum_request_count) - redirect_target_weights[redirect_target]
```

Comment:
what does this math here represent?

Reply:
This is because lua does not have negative numbers, this is just to get a reasonable metric for weight based on the heaviest